pkg/controller/pod_volume_restore_controller_test.go

```diff
@@ -0,0 +1,572 @@
+				Spec: arkv1api.PodVolumeRestoreSpec{
+					Pod: corev1api.ObjectReference{
+						Namespace: "ns-1",
+						Name:      "pod-1",
+					},
+				},
+				Status: arkv1api.PodVolumeRestoreStatus{
+					Phase: "",
+				},
+			},
+			pod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns-1",
+					Name:      "pod-1",
+				},
+				Spec: corev1api.PodSpec{
+					NodeName: "some-other-node",
+					InitContainers: []corev1api.Container{
+						{
+							Name: restic.InitContainer,
+						},
+					},
+				},
+				Status: corev1api.PodStatus{
+					InitContainerStatuses: []corev1api.ContainerStatus{
+						{
+							State: corev1api.ContainerState{
+								Running: &corev1api.ContainerStateRunning{
+									StartedAt: metav1.Time{Time: time.Now()},
+								},
+							},
+						},
+					},
+				},
+			},
+			shouldEnqueue: false,
+		},
+		{
+			name: "Empty phase pvr with pod on node not running init container should not be enqueued",
+			obj: &arkv1api.PodVolumeRestore{
+				Spec: arkv1api.PodVolumeRestoreSpec{
+					Pod: corev1api.ObjectReference{
+						Namespace: "ns-1",
+						Name:      "pod-1",
+					},
+				},
+				Status: arkv1api.PodVolumeRestoreStatus{
+					Phase: "",
+				},
+			},
+			pod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns-1",
+					Name:      "pod-1",
+				},
+				Spec: corev1api.PodSpec{
+					NodeName: controllerNode,
+					InitContainers: []corev1api.Container{
+						{
+							Name: restic.InitContainer,
+						},
+					},
+				},
+				Status: corev1api.PodStatus{
+					InitContainerStatuses: []corev1api.ContainerStatus{
+						{
+							State: corev1api.ContainerState{},
+						},
+					},
+				},
+			},
+			shouldEnqueue: false,
+		},
+		{
+			name: "Empty phase pvr with pod on node running init container should be enqueued",
+			obj: &arkv1api.PodVolumeRestore{
+				Spec: arkv1api.PodVolumeRestoreSpec{
+					Pod: corev1api.ObjectReference{
+						Namespace: "ns-1",
+						Name:      "pod-1",
+					},
+				},
+				Status: arkv1api.PodVolumeRestoreStatus{
+					Phase: "",
+				},
+			},
+			pod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns-1",
+					Name:      "pod-1",
+				},
+				Spec: corev1api.PodSpec{
+					NodeName: controllerNode,
+					InitContainers: []corev1api.Container{
+						{
+							Name: restic.InitContainer,
+						},
+					},
+				},
+				Status: corev1api.PodStatus{
+					InitContainerStatuses: []corev1api.ContainerStatus{
+						{
+							State: corev1api.ContainerState{
+								Running: &corev1api.ContainerStateRunning{
+									StartedAt: metav1.Time{Time: time.Now()},
+								},
+							},
+						},
+					},
+				},
+			},
+			shouldEnqueue: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				client      = kubefake.NewSimpleClientset()
+				informers   = kubeinformers.NewSharedInformerFactory(client, 0)
+				podInformer = informers.Core().V1().Pods()
```

Comment:
You could construct the podInformer by hand instead of having the factory do it. Then you don't need the fake kube clientset.